### DISCUSSION TITLE
[Suggestion] How to prevent loader in favor of location.state instead ?

### DISCUSSION THREAD

**Discussion by Sharlaan** [remix-run/remix#3227] [Q&A]
In a classic architecture consisting in a list and "details" pages, and a edit/add form page, i'm facing the following case :

Given the parent page (items list) consumes only a fexw properties from item (usually id and name) : OK

... i can then access an item's details in dedicated page, using a loader with params.id from routing url : OK

BUT now if i want to edit that item, from its details page where i already have all its properties, i then navigate to page edit.$id.tsx, passing all item's properties to Link.state, and then 2 cases to consider :
1. consume useLocation = location.state with state = item (PS: and state's type = unknown, not cool...)
2. if i want to access this edit.$id.tsx page directly, (then state = undefined), i need to implement a loader for that, using params.id exact same loader as the $id page

Result: my edit.$id.tsx has a loader AND location.state to deal with ...

I did not find any property in loaderFunction args interface that would help me check if location.state is undefined, hen allow execution, return otherwise.

Is there another way i missed ? Using location.state from navigation instead of loader could potentially reduce request trips ?

```ts
export const loader: LoaderFunction = async ({ params, context, request }) => {
  console.log('Loader used !!!', { params, context, request });
  // Loader is always executed, even though useLoaderData is commented out ?
  const data = await supabase
    .from<Example>('Examples')
    .select('*')
    .eq('id', params.id as string)
    .single()
    .then(({ data }) => (data ? snakeToCamelObject(data) : null));

  if (!data) {
    throw json({ message: 'Example could not be found' }, 404);
  }

  return data;
};

export default function EditExample() {
  const { state: example } = useLocation();
  // TODO: make location.state priority hivgher than loader
  // const example = useLoaderData<Example>();

  return <ExampleForm example={example} />;
}
```

**Comment by kiliman** [comment]
Unless your loader is extremely expensive, I wouldn't worry about trying to eliminate an extra fetch. As your example shows, it adds additional complexity and you still have to handle the case of requesting the data anyway (for direct access to edit page).

Remix is all about simplifying DX and improving UX. For 90% of the cases, Remix does this out of the box. I would trust Remix to do the right thing until you have measured and know for sure you can improve it at the expense of DX suffering.

As in all things in life, there will be tradeoffs.
